test/e2e/network/dns_common.go

```diff
@@ -613,3 +613,35 @@ func generateDNSUtilsPod() *v1.Pod {
 		},
 	}
 }
+
+func getDNSDomain(f *framework.Framework) string {
```

Comment:
This seems to be overcomplicating things and I'm afraid this will introduce more flakiness.

Can we simply make cluster DNS domain configurable in test via flags?
https://github.com/kubernetes/kubernetes/blob/9c94363ef90dead717a7e1cb4466040543528b55/test/e2e/framework/test_context.go#L235-L255

Reply:
Agreed this seems like it should be a 1 time pad on the startup of the framework and leveraged by the framework.

Reply:
@MrHohn @timothysc I made the changes to make the DNS domain configurable in test.